src/legacy/core_plugins/metrics/public/components/splits/filter.js

```diff
@@ -61,16 +59,19 @@ export const SplitByFilter = props => {
             />
           }
         >
-          <QueryBarInput
-            query={{
-              language: model.filter.language || getDefaultQueryLanguage(),
-              query: model.filter.query || '',
-            }}
-            onChange={filter => onChange({ filter })}
-            appName={'VisEditor'}
-            indexPatterns={[indexPattern]}
-            store={localStorage}
-          />
+          <QueryInputBarContext.Consumer>
```

Comment:
@lukeelmers @lizozom are you sure you want to wrap all QueryBarInput components? 
can we use  **useContext** method inside QueryBarInput component?

Reply:
Implemented your suggestions.
Thanks!